compiler/rustc_ast/src/attr/mod.rs

```diff
@@ -328,7 +328,9 @@ pub fn mk_name_value_item_str(ident: Ident, str: Symbol, str_span: Span) -> Meta
 }
 
 pub fn mk_name_value_item(ident: Ident, kind: LitKind, lit_span: Span) -> MetaItem {
-    let lit = MetaItemLit { token_lit: kind.to_token_lit(), kind, span: lit_span };
+    let token_lit = kind.synthesize_token_lit();
```

Comment:
This use of `synthesize_token_lit` is suspicious, we don't need to go through a token if we are directly constructing a `MetaItemLit`, need to check where is this `mk_name_value_item` used.

Reply:
Removed.